prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -2201,6 +2223,70 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
         Ok(result)
     }
 
+    fn encode_slice_len_call(
+        &mut self,
+        destination: &Option<(mir::Place<'tcx>, BasicBlockIndex)>,
+        args: &[mir::Operand<'tcx>],
+        location: mir::Location,
+        span: Span,
+    ) -> SpannedEncodingResult<Vec<vir::Stmt>> {
+        assert!(args.len() == 1, "unexpected args to slice::len(): {:?}", args);
+        let slice_operand = self.mir_encoder.encode_operand_expr(&args[0])
+            .with_span(span)?;
+
+        let mut stmts = vec![];
+
+        // we need to put a label before, it seems..
+        let label = self.cfg_method.get_fresh_label_name();
+        stmts.push(vir::Stmt::Label(label.clone()));
+
+        let slice_ty_ref = self.mir_encoder.get_operand_ty(&args[0]);
+        let slice_ty = if let ty::TyKind::Ref(_, slice_ty, _) = slice_ty_ref.kind() { slice_ty } else { unreachable!() };
+        let st = self.encode_slice_types(slice_ty).with_span(span)?;
+        let slice_len_name = self.encoder.encode_builtin_function_use(
+            BuiltinFunctionKind::SliceLen {
+                slice_ty_pred: st.slice_pred.clone(),
+                elem_ty_pred: st.elem_pred,
+            }
+        );
+
+        let rhs = vir::Expr::func_app(
+            slice_len_name,
+            vec![
+                slice_operand.clone(),
```

Comment:
See the Clippy warning reported in Github's diff.

Reply:
adressed, i really like that way of showing the warnings :-)